connectors/services/job_scheduling.py

```diff
@@ -94,19 +95,8 @@ async def _schedule(self, connector):
                 validator=source_klass(connector.configuration)
             )
 
-        if not await self._should_sync(connector):
-            return
-
-        trigger_method = (
-            JobTriggerMethod.ON_DEMAND
-            if connector.sync_now
-            else JobTriggerMethod.SCHEDULED
-        )
-        await self.sync_job_index.create(
-            connector=connector, trigger_method=trigger_method
-        )
-        if connector.sync_now:
-            await connector.reset_sync_now_flag()
+        await self._on_demand_sync(connector)
+        await self._scheduled_sync(connector)
```

Comment:
Should this also be called `_on_scheduled_sync` for consistency?

Reply:
I think `on_demand` and `scheduled` here make sense - on is not an indicator of "event", but just a part of "on demand" word combination

Reply:
`on_demand` and `scheduled` are the two values of `JobTriggerMethod` Enum.
https://github.com/elastic/connectors-python/blob/947b8734fc48f3bc760e648487ff1944ba0c6501/connectors/byoc.py#L60-L63